ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/tracking/TrackerMap.scala

```diff
@@ -93,24 +91,25 @@ class TrackerMap(retentionPeriod: FiniteDuration, loggerFactory: NamedLoggerFact
 }
 
 object TrackerMap {
-  case class Key(application: String, party: String)
+
+  final case class Key(application: String, party: String)
+
+  sealed trait AsyncResourceState[+T <: AutoCloseable]
```

Comment:
Why did this trait become generic?

Reply:
It used to be part of `AsyncResource` and this "locked" the generic `T` correctly but caused a warning that was silenced. I did this to make sure `AsyncResource` can use it by explicitly assigning the type but without causing the warning (caused by [this](https://github.com/scala/bug/issues/4440)).